### DISCUSSION TITLE
이메일 인증 관련 토의

### DISCUSSION THREAD

**Discussion by senasoon** [when-where/whenwhere-server#11] [General]
이메일 인증 관련 토의할만한것 공유드립니다.
@imironjin @sykang013 

**1. 메일 발송 기업 설정 Naver vs Gmail**
- Naver 하루 1000건 이상?(너무 많으면 데이터 과금 주의라 하네요)
- Gmail 하루 500건

:arrow_right: 제가 지금 Naver 새로 가입이 불가해서 **google에 새로 가입해서 사용중입니다.**

**2. 이메일 인증 토큰 만료 및 데이터베이스 삭제**
현재 이메일 인증은 완료된 상태입니다. 
이메일 인증 절차는 다음과 같습니다.

회원가입 완료 및 데이터 생성 > 이메일 인증 완료 > 로그인 가능

회원가입은 되지만 이메일 인증이 완료되지 않은 경우 로그인이 불가능합니다.
데이터베이스에 인증이 됐는지 안됐는지에 따라 is_valid 값이 주어지는데요. 
만약 인증을 하지 않으면 데이터 베이스에 가입이 완전히 되지 않은 데이터들이 계속 남아 있는데 주기적으로 이런 데이터를 삭제해야할까요?
삭제하게 된다면 이메일 인증 토큰도 만료 시간이 설정되어야 할 것 같습니다.

**Comment by sykang013** [comment]
1. 공식 메일은 gmail을 사용하는게 좋아 보입니다. 
2. 데이터는 주기적으로 삭제해야할것 같습니다. 저는 3일 정도로 좀 짧아도 될 것 같다고 생각합니다.
- 대신 '이메일 인증을 3일 이내에 완료해야만 회원가입이 완료됩니다.' 를 명시해줘도 좋을 것 같습니다.

**Comment by imironjin** [comment]
1. 하루 사용량 말고 그 외 다른점이 있는지 궁금합니다.
2. 불필요한 데이터들은 주기적으로 삭제하는게 좋을 것 같습니다. 주기는 의논해보면 될 것 같습니다.

**Reply by senasoon** [reply]
하루 사용량 말고 다른 차이점은 잘 모르겠습니다!

**Comment by senasoon** [comment]
주기를 1일로 하는건 어떠신가요? 스케쥴러가 매일 자정에 만료된 토큰들을 삭제하는 방식으로 하고 토큰 생성할 때 expire를 하루로 하겠습니다!

**Reply by imironjin** [reply]
좋은 것 같습니다.

**Reply by sykang013** [reply]
좋아요!
